### ISSUE TITLE
Notice: Undefined offset: 0

### ISSUE THREAD

**Issue by i-gabdushev** [WhichBrowser/Parser-PHP#86]
Specific user agent string produces error
```
$s = "Mozilla/5.0 (Linux; Android 4.2.2; KENEKSI Build/JDQ39; ru-ru) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Mobile Safari/537.36 Puffin/6.1.4.16005AP";
$result = new \WhichBrowser\Parser($s);
```

**Comment by NielsLeenheer** [comment]
Which version of PHP are you using and could you post the full error message, including the file named line number in which the error occurs.

**Comment by i-gabdushev** [comment]
tried these versions
- PHP 7.1.10
- PHP 5.6.30

```
PHP Notice:  Undefined offset: 0 in vendor/whichbrowser/parser/src/Data/DeviceModels.php on line 227
PHP Notice:  Undefined offset: 1 in vendor/whichbrowser/parser/src/Data/DeviceModels.php on line 228
```
also stack trace might be useful
```
PHP Notice:  Undefined offset: 0 in /my_project/vendor/whichbrowser/parser/src/Data/DeviceModels.php on line 227
PHP Stack trace:
PHP   1. {main}() /my_project/src/t.php:0
PHP   2. WhichBrowser\Parser->__construct() /my_project/src/t.php:4
PHP   3. WhichBrowser\Parser->analyse() /my_project/vendor/whichbrowser/parser/src/Parser.php:23
PHP   4. WhichBrowser\Analyser->analyse() /my_project/vendor/whichbrowser/parser/src/Parser.php:56
PHP   5. WhichBrowser\Analyser->analyseHeaders() /my_project/vendor/whichbrowser/parser/src/Analyser.php:42
PHP   6. WhichBrowser\Analyser->analyseUserAgent() /my_project/vendor/whichbrowser/parser/src/Analyser/Header.php:15
PHP   7. WhichBrowser\Analyser\Header\Useragent->__construct() /my_project/vendor/whichbrowser/parser/src/Analyser/Header.php:85
PHP   8. WhichBrowser\Analyser\Header\Useragent->detectOperatingSystem() /my_project/vendor/whichbrowser/parser/src/Analyser/Header/Useragent.php:19
PHP   9. WhichBrowser\Analyser\Header\Useragent->detectAndroid() /my_project/vendor/whichbrowser/parser/src/Analyser/Header/Useragent/Os.php:19
PHP  10. WhichBrowser\Data\DeviceModels::identify() /my_project/vendor/whichbrowser/parser/src/Analyser/Header/Useragent/Os.php:311
PHP  11. WhichBrowser\Data\DeviceModels::identifyAndroid() /my_project/vendor/whichbrowser/parser/src/Data/DeviceModels.php:55
PHP  12. WhichBrowser\Data\DeviceModels::identifyList() /my_project/vendor/whichbrowser/parser/src/Data/DeviceModels.php:167

```

**Comment by NielsLeenheer** [comment]
Found the issue and will be fixed in the next release.

**Comment by NielsLeenheer** [comment]
Fixed in v2.0.29
